libs/usb-drive/src/mocks/file_usb_drive.test.ts

```diff
@@ -0,0 +1,56 @@
+import { Buffer } from 'buffer';
+import { existsSync } from 'fs';
+import { join } from 'path';
+import {
+  MOCK_USB_FILE_PATH,
+  MockFileUsbDrive,
+  getMockFileUsbDriveHandler,
+  initializeMockFile,
+} from './file_usb_drive';
+
+beforeEach(() => {
+  // clear mock file from other tests or development
+  initializeMockFile();
```

Comment:
Yeah, there are a lot of disadvantages to the fact that, in this first pass, all file mocks in all use cases use the same temporary file. It would be better to introduce the idea of "different instances." And it's not currently set up to replace our dev USB drive mocks, although we will get there.

The single file approach solves the problem of multiple processes (e.g. app and integration test runner) being able to access / edit the same data. This problem is a little different in the integration test case, the normal test case, and development. 

After doing this PR, I think that the integration test case could be solved by environment variables? As in, the test suite generates a UUID on boot up and places it in an environment variable, which can then be used to initialize the `usbDrive` from the app, so they can access the same temporary file. The other two cases - regular tests and dev - are easier. In tests we could mock the environment variable and in dev (`NODE_ENV=development`) we could actually have a `git` ignored file and mock USB root directory under `libs/usb-drive/dev-workspace`. Then we could kill `libs/usb-mocking`. What do you think of these ideas?

Reply:
> there are a lot of disadvantages to the fact that, in this first pass, all file mocks in all use cases use the same temporary file

I think we should get clear on what the disadvantages are before we try to solve this. I'm thinking about the auth mocks, which I believe are backed by a single file instance, and it seems like that hasn't caused too many issues so far.

I do like the idea of killing `libs/usb-mocking`, but I would imagine that's orthogonal to this discussion, since we only need a single mock instance for that.

Reply:
I agree. I started to solve this in the most general way and found it to be more trouble than it's worth.

In a more limited way, however, it's necessary - distinguishing the dev USB from the test USB. We're more concerned about data persistence in this case than in the auth case. I wouldn't want my integration tests clearing a dev mock USB drive I was separately using. This will be addressed in another PR not up for review yet.

Reply:
That makes sense!